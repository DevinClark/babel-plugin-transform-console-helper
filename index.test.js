var plugin = require('./');
var { transform } = require('babel-core');

it('works', () => {
  var example = `
    console.log(a);
    console.warn(a);
    console.info(a);
    console.error(a);
  `;
  const { code } = transform(example, {
    plugins: [plugin],
  });

  expect(code).toMatchSnapshot();
});

it('works with a parent', () => {
  var example = `
  function onChange() {
    console.log(...things, a, b,'c', 4);
    console.warn(...things, a, b,'c', 4);
    console.info(...things, a, b,'c', 4);
    console.error(...things, a, b,'c', 4);
  }
  `;

  const { code } = transform(example, {
    plugins: [plugin],
  });

  expect(code).toMatchSnapshot();
});

it('does not change non-console logging functions', () => {
  var example = `
  function onChange(a) {
  }

  onChange(a);
  window.setTimeout(onChange, 500);
  console.group(a);
  console.groupEnd(a);
  `;

  const { code } = transform(example, {
    plugins: [plugin],
  });

  expect(code).toMatchSnapshot();
});
