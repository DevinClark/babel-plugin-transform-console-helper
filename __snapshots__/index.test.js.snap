// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change non-console logging functions 1`] = `
"
function onChange(a) {}

onChange(a);
window.setTimeout(onChange, 500);
console.group(a);
console.groupEnd(a);"
`;

exports[`works 1`] = `
"
console.log(\\"a\\", a);
console.warn(\\"a\\", a);
console.info(\\"a\\", a);
console.error(\\"a\\", a);"
`;

exports[`works with a parent 1`] = `
"
function onChange() {
  console.log('onChange()', ...things, 'a', a, 'b', b, 'c', 4);
  console.warn('onChange()', ...things, 'a', a, 'b', b, 'c', 4);
  console.info('onChange()', ...things, 'a', a, 'b', b, 'c', 4);
  console.error('onChange()', ...things, 'a', a, 'b', b, 'c', 4);
}"
`;
